be/src/olap/tablet_manager.cpp

```diff
@@ -941,11 +946,15 @@ OLAPStatus TabletManager::report_all_tablets_info(std::map<TTabletId, TTablet>*
             TTabletInfo tablet_info;
             tablet_ptr->build_tablet_report_info(&tablet_info);
 
-            // report expire transaction
+            // find expire transaction corresponding to this tablet
+            TabletInfo tinfo = TabletInfo(tablet_ptr->tablet_id(), tablet_ptr->schema_hash(), tablet_ptr->tablet_uid());
             vector<int64_t> transaction_ids;
-            // TODO(ygl): tablet manager and txn manager may be dead lock
-            StorageEngine::instance()->txn_manager()->get_expire_txns(tablet_ptr->tablet_id(), 
-                tablet_ptr->schema_hash(), tablet_ptr->tablet_uid(), &transaction_ids);
+            auto find = expire_txn_map.find(tinfo);
+            if (find != expire_txn_map.end()) {
+                for(auto& it : find->second) {
+                    transaction_ids.push_back(it);
+                }
+            }
```

Comment:
_clear_txn_partition_map_unlocked() in delete_txn() may be better to modified together.